### PR TITLE
Feature/accept request screen upgrade

### PR DESCRIPTION
**Pull request by ErwanGuesry**
<html>
<body>
<h1 data-start="144" data-end="153">Summary</h1>
<p data-start="155" data-end="442">This Pull Request introduces a complete refactor and enhancement of the <strong data-start="227" data-end="245">Accept Request</strong> feature in the Android application.<br data-start="281" data-end="284">
The goal was to improve <strong data-start="308" data-end="324">UI structure</strong>, <strong data-start="326" data-end="347">code organization</strong>, <strong data-start="349" data-end="369">state management</strong>, and <strong data-start="375" data-end="392">test coverage</strong> for better maintainability and user experience.</p>
<p data-start="444" data-end="676">Additionally, this work was <strong data-start="472" data-end="526">partially assisted by Artificial Intelligence (AI)</strong> for drafting and refactoring sections of code and documentation.<br data-start="591" data-end="594">
All generated content was reviewed and validated by a developer before submission.</p>
<hr data-start="678" data-end="681">
<h1 data-start="683" data-end="700">What's Included</h1>
<h3 data-start="702" data-end="725">Core Implementation</h3>
<ul data-start="726" data-end="1460">
<li data-start="726" data-end="1077">
<p data-start="728" data-end="809"><strong data-start="728" data-end="751">AcceptRequestScreen</strong> has been redesigned for better structure and readability:</p>
<ul data-start="812" data-end="1077">
<li data-start="812" data-end="905">
<p data-start="814" data-end="905">Introduced composables <code data-start="837" data-end="855">RequestDetailRow</code> and <code data-start="860" data-end="876">CreatorSection</code> for modular and reusable UI.</p>
</li>
<li data-start="908" data-end="1010">
<p data-start="910" data-end="1010">Implemented <code data-start="922" data-end="952">AcceptRequestScreenConstants</code> to centralize spacing, typography, and layout dimensions.</p>
</li>
<li data-start="1013" data-end="1077">
<p data-start="1015" data-end="1077">Improved layout hierarchy with cards, icons, and text styling.</p>
</li>
</ul>
</li>
<li data-start="1078" data-end="1339">
<p data-start="1080" data-end="1116">Enhanced <strong data-start="1089" data-end="1115">AcceptRequestViewModel</strong>:</p>
<ul data-start="1119" data-end="1339">
<li data-start="1119" data-end="1194">
<p data-start="1121" data-end="1194">Added support for fetching creator profiles from <code data-start="1170" data-end="1193">UserProfileRepository</code>.</p>
</li>
<li data-start="1197" data-end="1272">
<p data-start="1199" data-end="1272">Improved state handling with explicit loading, success, and error states.</p>
</li>
<li data-start="1275" data-end="1339">
<p data-start="1277" data-end="1339">Reversed the logic of accept/cancel functions for correctness.</p>
</li>
</ul>
</li>
<li data-start="1340" data-end="1460">
<p data-start="1342" data-end="1460">Updated <strong data-start="1350" data-end="1378">AcceptRequestScreenTests</strong> with new and extended UI test cases covering multiple states and edge conditions.</p>
</li>
</ul>
<h3 data-start="1462" data-end="1503">State Management / Logic Improvements</h3>
<ul data-start="1504" data-end="1778">
<li data-start="1504" data-end="1597">
<p data-start="1506" data-end="1597">Introduced new <code data-start="1521" data-end="1534">creatorName</code> field in <code data-start="1544" data-end="1566">AcceptRequestUIState</code> to display author information.</p>
</li>
<li data-start="1598" data-end="1673">
<p data-start="1600" data-end="1673">Added robust error handling for failed data fetches or repository issues.</p>
</li>
<li data-start="1674" data-end="1778">
<p data-start="1676" data-end="1778">Refactored <code data-start="1687" data-end="1702">acceptRequest</code> and <code data-start="1707" data-end="1734">cancelAcceptanceToRequest</code> functions for clearer flow and reliability.</p>
</li>
</ul>
<h3 data-start="1780" data-end="1809">Code Quality Enhancements</h3>
<ul data-start="1810" data-end="2116">
<li data-start="1810" data-end="1894">
<p data-start="1812" data-end="1894">Introduced <code data-start="1823" data-end="1853">AcceptRequestScreenConstants</code> file for centralized styling parameters.</p>
</li>
<li data-start="1895" data-end="1954">
<p data-start="1897" data-end="1954">Replaced hardcoded spacing and text sizes with constants.</p>
</li>
<li data-start="1955" data-end="2053">
<p data-start="1957" data-end="2053">Improved composable decomposition, naming clarity, and alignment with Material 3 best practices.</p>
</li>
<li data-start="2054" data-end="2116">
<p data-start="2056" data-end="2116">Enhanced log tracing for debugging and better observability.</p>
</li>
</ul>
<hr data-start="2118" data-end="2121">
<h1 data-start="2123" data-end="2147">Testing Implementation</h1>
<h3 data-start="2149" data-end="2172">Test Infrastructure</h3>
<ul data-start="2173" data-end="2554">
<li data-start="2173" data-end="2460">
<p data-start="2175" data-end="2237">Extended existing UI test suite in <code data-start="2210" data-end="2236">AcceptRequestScreenTests</code>:</p>
<ul data-start="2240" data-end="2460">
<li data-start="2240" data-end="2331">
<p data-start="2242" data-end="2331">Added tests for button states, back navigation, loading indicator, and archived requests.</p>
</li>
<li data-start="2334" data-end="2390">
<p data-start="2336" data-end="2390">Validated tag display, creator name, and empty states.</p>
</li>
<li data-start="2393" data-end="2460">
<p data-start="2395" data-end="2460">Verified proper UI transitions between accept and cancel actions.</p>
</li>
</ul>
</li>
<li data-start="2461" data-end="2554">
<p data-start="2463" data-end="2554">Achieved high coverage across <strong data-start="2493" data-end="2504">loading</strong>, <strong data-start="2506" data-end="2517">success</strong>, <strong data-start="2519" data-end="2528">error</strong>, and <strong data-start="2534" data-end="2546">archived</strong> states.</p>
</li>
</ul>
<h3 data-start="2556" data-end="2581">State / Logic Testing</h3>
<ul data-start="2582" data-end="2842">
<li data-start="2582" data-end="2842">
<p data-start="2584" data-end="2603">Tests now validate:</p>
<ul data-start="2606" data-end="2842">
<li data-start="2606" data-end="2660">
<p data-start="2608" data-end="2660">Dynamic text changes on accept/cancel button clicks.</p>
</li>
<li data-start="2663" data-end="2708">
<p data-start="2665" data-end="2708">Correct behavior for unauthenticated users.</p>
</li>
<li data-start="2711" data-end="2767">
<p data-start="2713" data-end="2767">Proper error message rendering and visual consistency.</p>
</li>
<li data-start="2770" data-end="2842">
<p data-start="2772" data-end="2842">Multiple request types and tag list rendering (including empty cases).</p>
</li>
</ul>
</li>
</ul>
<hr data-start="2844" data-end="2847">
<h1 data-start="2849" data-end="2872">Screenshots / Visuals</h1>
image
image
</blockquote>
<hr data-start="2983" data-end="2986">
<h1 data-start="2988" data-end="3022">What's Not Included (Future PRs)</h1>
<ul data-start="3024" data-end="3276">
<li data-start="3075" data-end="3132">
<p data-start="3077" data-end="3132">Integration with backend analytics or error tracking.</p>
</li>
<li data-start="3133" data-end="3216">
<p data-start="3135" data-end="3216">Additional accessibility improvements (TalkBack labels, contrast verification).</p>
</li>
<li data-start="3217" data-end="3276">
<p data-start="3219" data-end="3276">End-to-end instrumentation tests for real Firestore data.</p>
</li>
</ul>
<hr data-start="3278" data-end="3281">
<h1 data-start="3283" data-end="3307">Technical Improvements</h1>
<div class="_tableContainer_1rjym_1"><div tabindex="-1" class="group _tableWrapper_1rjym_13 flex w-fit flex-col-reverse">
Area | Improvement | Benefit
-- | -- | --
UI Architecture | Introduced modular composables (RequestDetailRow, CreatorSection) | Better maintainability and reusability
State Management | Added creatorName and improved request loading | Clearer and more informative UI
Testing | Expanded UI tests to cover new cases | Improved reliability and regression prevention
Code Quality | Centralized constants and reduced duplication | Easier to adjust UI globally
Logging | Added detailed debug logs for ViewModel flows | Simplified troubleshooting

</div></div>
<hr data-start="3902" data-end="3905">
<h1 data-start="3907" data-end="3914">Notes</h1>
<p data-start="3916" data-end="4262">This PR provides a strong foundation for future UI enhancements and backend integrations.<br data-start="4005" data-end="4008">
By restructuring both <strong data-start="4030" data-end="4036">UI</strong> and <strong data-start="4041" data-end="4061">ViewModel layers</strong>, the Accept Request feature becomes more scalable, testable, and maintainable.<br data-start="4140" data-end="4143">
Future iterations can focus on <strong data-start="4174" data-end="4195">styling alignment</strong>, <strong data-start="4197" data-end="4225">accessibility compliance</strong>, and <strong data-start="4231" data-end="4259">real API synchronization</strong>.</p>
<p data-start="4264" data-end="4303">AI tools were leveraged to assist with:</p>
<ul data-start="4304" data-end="4496">
<li data-start="4304" data-end="4362">
<p data-start="4306" data-end="4362">Drafting composable structures and naming conventions.</p>
</li>
<li data-start="4363" data-end="4405">
<p data-start="4365" data-end="4405">Generating the initial test templates.</p>
</li>
<li data-start="4406" data-end="4496">
<p data-start="4408" data-end="4496">Suggesting documentation formatting.<br data-start="4444" data-end="4447">
All code has been reviewed and adjusted manually.</p>
</li>
</ul>
<hr data-start="4498" data-end="4501">
<p data-start="4503" data-end="4516"><strong data-start="4503" data-end="4516">Checklist</strong></p>
<ul class="contains-task-list" data-start="4517" data-end="4695">
<li class="task-list-item" data-start="4517" data-end="4556">
<p data-start="4523" data-end="4556"><input disabled="" type="checkbox" checked=""> Feature implementation complete</p>
</li>
<li class="task-list-item" data-start="4557" data-end="4596">
<p data-start="4563" data-end="4596"><input disabled="" type="checkbox" checked=""> Unit/UI tests added and passing</p>
</li>
<li data-start="4597" data-end="4654" class="task-list-item">
<p data-start="4603" data-end="4654"><input disabled="" type="checkbox" checked=""> Code reviewed for maintainability and readability</p>
</li>
<li data-start="4655" data-end="4695" class="task-list-item">
<p data-start="4661" data-end="4695"><input disabled="" type="checkbox" checked=""> Documentation / comments updated</p>
</li>
</ul>
<hr data-start="4697" data-end="4700">
<p data-start="4702" data-end="4825"><strong data-start="4702" data-end="4711">Type:</strong> <code data-start="4712" data-end="4748">feature/accept-request-ui-refactor</code><br data-start="4748" data-end="4751">
<strong data-start="4751" data-end="4761">Scope:</strong> <code data-start="4762" data-end="4793">Overview / Request Management</code><br data-start="4793" data-end="4796">
<strong data-start="4796" data-end="4816">Issue Reference:</strong> <code data-start="4817" data-end="4823">#163</code></p>
</body>
</html>

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/overview/AcceptRequest.kt [hunk 1]
```diff
@@ -97,52 +115,84 @@ fun AcceptRequestScreen(
...
+                            RequestDetailRow(
+                                icon = Icons.Outlined.Notifications,
+                                label = "Status",
+                                content = request.status.displayString(),
+                                testTag = AcceptRequestScreenTestTags.REQUEST_STATUS)
+
+                            // Location
+                            RequestDetailRow(
+                                icon = Icons.Outlined.LocationOn,
+                                label = "Location",
```
Comment (Thibaud-Babin) [inline app/src/main/java/com/android/sample/ui/overview/AcceptRequest.kt]: Magic value

File: app/src/main/java/com/android/sample/ui/overview/AcceptRequest.kt [hunk 2]
```diff
@@ -97,52 +115,84 @@ fun AcceptRequestScreen(
...
+                                icon = Icons.Outlined.BookmarkBorder,
+                                label = "Request type",
+                                content =
+                                    request.requestType.joinToString(", ") { it.displayString() },
+                                testTag = AcceptRequestScreenTestTags.REQUEST_TYPE)
+
+                            // Status
+                            RequestDetailRow(
+                                icon = Icons.Outlined.Notifications,
+                                label = "Status",
```
Comment (Thibaud-Babin) [inline app/src/main/java/com/android/sample/ui/overview/AcceptRequest.kt]: magic value

File: app/src/main/java/com/android/sample/ui/overview/AcceptRequest.kt [hunk 3]
```diff
@@ -97,52 +115,84 @@ fun AcceptRequestScreen(
...
+                                label = "Description",
+                                content = request.description,
+                                testTag = AcceptRequestScreenTestTags.REQUEST_DESCRIPTION)
 
-                Spacer(modifier = Modifier.height(24.dp))
+                            // Tags
+                            RequestDetailRow(
+                                icon = Icons.Outlined.LocalOffer,
+                                label = "Tags",
+                                content = request.tags.joinToString(", ") { it.displayString() },
```
Comment (Thibaud-Babin) [inline app/src/main/java/com/android/sample/ui/overview/AcceptRequest.kt]: magic value

File: app/src/main/java/com/android/sample/ui/overview/AcceptRequestViewModel.kt [hunk 4]
```diff
@@ -83,19 +114,19 @@ class AcceptRequestViewModel(
...
   /**
-   * Cancel the acceptance of the request, and update the UI state
+   * Accept the request by its id, and update the UI state
    *
-   * @param requestID the id of the request to cancel
+   * @param requestID the id of the request to accept
    */
-  fun cancelAcceptanceToRequest(requestID: String) {
+  fun acceptRequest(requestID: String) {
     viewModelScope.launch {
```
Comment (Thibaud-Babin) [inline app/src/main/java/com/android/sample/ui/overview/AcceptRequestViewModel.kt]: Again, logging is good  but consider adding a verbose mode maybe . Otherwise this implementations looks good !

File: app/src/main/java/com/android/sample/ui/overview/AcceptRequestViewModel.kt
Comment (Thibaud-Babin) [inline app/src/main/java/com/android/sample/ui/overview/AcceptRequestViewModel.kt]: This update provides better logging, more resilient error handling, and an overall clearer logic flow between accepting and canceling a request. The decision to fetch and include the creator’s display name adds value to the UI while ensuring fallback behavior when the repository is null or the fetch fails. However, the logging verbosity may be excessive for production and could benefit from a conditional debug flag or structured logging via a library like Timber. The method naming is mostly clear, though cancelAcceptanceToRequest could be shortened to cancelRequestAcceptance() for readability. The use of immutable state updates with .copy() is clean and idiomatic. Overall, this file demonstrates a mature handling of asynchronous state management with minor opportunities for polish.

File: app/src/androidTest/java/com/android/sample/ui/acceptRequest/AcceptRequestScreenTests.kt
Comment (Thibaud-Babin) [inline app/src/androidTest/java/com/android/sample/ui/acceptRequest/AcceptRequestScreenTests.kt]: The test suite provides solid coverage of the main user interactions and UI states for AcceptRequestScreen. It effectively validates critical behaviors such as navigation via the back button, correct display of multiple request types and statuses, and handling of edge cases like empty tag lists. The use of testTags makes the tests reliable and easy to maintain, while the structure follows Compose testing best practices.

That said, a few minor improvements could strengthen the tests further: consider asserting specific loading indicators in buttonShowsLoadingState() instead of only checking existence, and use assertTrue or Truth assertions instead of raw assert() for clearer failure messages. Overall, the tests are concise, readable, and provide strong functional validation of the screen’s behavior.

File: app/src/main/java/com/android/sample/ui/overview/AcceptRequest.kt [hunk 5]
```diff
@@ -152,24 +202,151 @@ fun AcceptRequestScreen(
...
+                  style = MaterialTheme.typography.titleMedium,
+                  fontWeight = FontWeight.SemiBold,
+                  color = MaterialTheme.colorScheme.primary)
+            }
+
+        Spacer(modifier = Modifier.width(AcceptRequestScreenConstants.CREATOR_AVATAR_TEXT_SPACING))
+
+        Column {
+          Text(
+              text = "Posted by",
```
Comment (Thibaud-Babin) [inline app/src/main/java/com/android/sample/ui/overview/AcceptRequest.kt]: Magic value
